### PR TITLE
Accumulated updates for heap profiling

### LINKED ISSUE #3236: TiFlash get stuck occasionally when mem-profiling is actived
Branch: https://github.com/JaySon-Huang/tics/tree/add_mem_prof

Start a TiFlash compiled with `-DENABLE_JEMALLOC_PROF=1`, and start tiflash with env `export MALLOC_CONF="prof:true,prof_active:true,lg_prof_interval:33"`, TiFlash get stuck occasionally. The stacks are as below:
[stuck-stacks-log.tar.gz](https://github.com/pingcap/tics/files/7352662/stuck-stacks-log.tar.gz)

### LINKED ISSUE #9241: KVStore & Proxy infra issue tracker

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What problem does this PR solve?

Issue Number: close #9241 close #3236

Problem Summary:

### What is changed and how it works?

```commit-message

```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: libs/libcommon/cmake/find_jemalloc.cmake [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 option (ENABLE_JEMALLOC "Set to TRUE to use jemalloc" ON)
 # TODO: Make ENABLE_JEMALLOC_PROF default value to ON after https://github.com/pingcap/tics/issues/3236 get fixed.
-option (ENABLE_JEMALLOC_PROF "Set to ON to enable jemalloc profiling" OFF)
+option (ENABLE_JEMALLOC_PROF "Set to ON to enable jemalloc profiling" ON)
```
Comment (Lloyd-Pottiger) [inline libs/libcommon/cmake/find_jemalloc.cmake:17]: Have we fixed this?

Comment (CalvinNeo) [reply libs/libcommon/cmake/find_jemalloc.cmake:17]: It does not related to the former deadlock.

BTW, the former deadlock has not appeared again so far.

Comment (Lloyd-Pottiger) [reply libs/libcommon/cmake/find_jemalloc.cmake:17]: Could we remove the comment?

Comment (CalvinNeo) [reply libs/libcommon/cmake/find_jemalloc.cmake]: fixed

**Review by Lloyd-Pottiger — APPROVED** [review]

**Comment by CalvinNeo** [comment]
mv: can't rename 'log': No such file or directory

**Review by breezewish — APPROVED** [review]

**Pull request merged**